pkg/controllers/provisioning/scheduling/volumetopology.go

```diff
@@ -181,15 +182,32 @@ func (v *VolumeTopology) ValidatePersistentVolumeClaims(ctx context.Context, pod
 				continue
 			}
 
-			storageClassName = pvc.Spec.StorageClassName
+			storageClassName = lo.FromPtr(pvc.Spec.StorageClassName)
```

Comment:
Is there ever a case where this should be empty? Why do we need to convert it from a pointer?

Reply:
There's a couple of cases:
1) There was no storage class specified and the default hasn't been applied yet by the persistent volume controller
2) There was no storage class specified and there is no default storage class

Reply:
This may not be the case, I found out there's a [default storage class admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass) which will populate the default. I'd have to take a closer look to see what its behavior is when there is no default storage class but this eliminates the first case at least. Not sure if we can rely on the presence of that admission controller but its compiled into the API server so it seems like a safe bet?

Reply:
Seems like the second case is still possible, the admission controller won't update the PVC if there is no default storage class ([ref](https://github.com/kubernetes/kubernetes/blob/8509ab82b96caa2365552efa08c8ba8baf11c5ec/plugin/pkg/admission/storage/storageclass/setdefault/admission.go#L112)). Makes sense, the persistent volume controller will also add the default storage class if one exists so that probably covers the case of a default being created after the PVC.